internal/pkg/expanders/storageBlob.go

```diff
@@ -0,0 +1,682 @@
+
+	if err != nil {
+		return ExpanderResult{
+			Err:               fmt.Errorf("Error getting blob: %s", err),
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+
+	result := string(buf)
+	return ExpanderResult{
+		Response:          ExpanderResponse{Response: result, ResponseType: ResponseXML},
+		SourceDescription: "StorageBlobExpander request",
+		Nodes:             []*TreeNode{},
+		IsPrimaryResponse: true,
+	}
+
+}
+
+func (e *StorageBlobExpander) deleteBlob(ctx context.Context, currentItem *TreeNode) (bool, error) {
+
+	containerID := currentItem.Metadata["ContainerID"]
+	containerName := e.getContainerName(containerID)
+	accountName := currentItem.Metadata["AccountName"]
+	accountKey := currentItem.Metadata["AccountKey"]
+	blobName := currentItem.Metadata["BlobName"]
+
+	blobEndpoint, err := e.getStorageBlobEndpoint(ctx, containerID)
+	if err != nil {
+		err = fmt.Errorf("Error getting blob endpoint: %s", err)
+		return false, err
+	}
+
+	// DeleteBlob docs: https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob
+	url := blobEndpoint + containerName + "/" + blobName
+	_, err = e.doRequest(ctx, "DELETE", url, accountName, accountKey, "/"+accountName+"/"+containerName+"/"+blobName)
+
+	if err != nil {
+		return false, fmt.Errorf("Error deleting blob: %s", err)
+	}
+
+	return true, nil
+
+}
+
+func (e *StorageBlobExpander) getAccountName(containerID string) string {
+	i := strings.Index(containerID, "/blobServices")
+	accountID := containerID[0:i]
+	i = strings.LastIndex(accountID, "/")
+	return accountID[i+1:]
+}
+func (e *StorageBlobExpander) getContainerName(containerID string) string {
+	i := strings.LastIndex(containerID, "/")
+	containerName := containerID[i+1:]
+	i = strings.Index(containerName, "?")
+	if i >= 0 {
+		containerName = containerName[:i] // strip query string
+	}
+	return containerName
+}
+
+func (e *StorageBlobExpander) getAccountKey(ctx context.Context, containerID string) (string, error) {
+
+	i := strings.Index(containerID, "/blobServices")
+	rootURL := containerID[0:i]
+	listKeysURL := rootURL + "/listKeys?api-version=2019-06-01"
+
+	data, err := e.armClient.DoRequest(ctx, "POST", listKeysURL)
+	if err != nil {
+		return "", fmt.Errorf("Error calling listKeys: %s", err)
+	}
+	response := StorageListKeyResponse{}
+	err = json.Unmarshal([]byte(data), &response)
+	if err != nil {
+		err = fmt.Errorf("Error unmarshalling response: %s\nURL:%s", err, listKeysURL)
+		return "", err
+	}
+	if len(response.Keys) == 0 {
+		err = fmt.Errorf("No keys in response: %s", err)
+		return "", err
+	}
+
+	return response.Keys[0].Value, nil
+}
+func (e *StorageBlobExpander) getStorageBlobEndpoint(ctx context.Context, containerID string) (string, error) {
+
+	i := strings.Index(containerID, "/blobServices")
+	rootURL := containerID[0:i]
+	storageAccountURL := rootURL + "?api-version=2019-06-01"
+
+	data, err := e.armClient.DoRequest(ctx, "GET", storageAccountURL)
```

Comment:
Similar note on caching, not sure how relevant it is, also might be something to track in an issue and tackle as a separate PR.

Reply:
Let's see how it goes with a bit of usage and address if needed?